particle-iot/particle-cli
# Logic Functions Implementation

## Contributors

@keeramis 
@hugomontero 

## Description

This PR adds support for Logic Functions in the CLI.

This is an implementation in it's initial phase that runs Logic Functions on a fundemental level

## How to Test

### Setup your CLI repo to this branch

1. Pull down branch: `git pull` && `git checkout feature/logic-function`
2. Run `nvm use` which currently selects node version 16
3. Install dependencies: npm install

### Run Logic Function commands

Run `npm start -- logic-function` or `npm start -- lf` for the list of commands.
Test away! 
- Create a Logic Function and deploy it
- Download an existing Logic Function to your disk, modify it locally, and deploy it
- Enable and disable Logic Functions
- Ensure everything works as expected and the UI looks helpful

### Instructions to edit Logic Functions in an IDE
https://www.notion.so/particle/Logic-Dev-Tools-Setup-e2722876e21340c0a1fa2564af53ad41


## Related Issues / Discussions

EPIC: https://app.shortcut.com/particle/epic/123383/logic-dev-tools-implementation

## Completeness
